### PR TITLE
Fix bug with as.character with priors section

### PR DESCRIPTION
**Pull request by G-Kazantzidis**
https://github.com/Genentech/jmpost/issues/45#issue-1327236040

Depends on main, add_os_model and fix_as.character_bug (commit ed293eb00c653e52d3920308e4cf6d416c1247e2) branches.

### CODE REVIEW

**Comment by gowerc** [comment]
@G-Kazantzidis , looks like the unit tests are failing, please can you resolve them

File: R/stan_module.R [hunk 1]
```diff
@@ -49,17 +49,15 @@ read_stan <- function(string) {
...
-        .Object,
-        ...,
-        functions = "",
-        data = "",
-        parameters = "",
-        transformed_parameters = "",
-        generated_quantities = "",
-        priors = list(),
-        inits = list()
-    ) {
```
Comment (gowerc) [inline R/stan_module.R:62]: why has this been changed ?

Comment (G-Kazantzidis) [reply R/stan_module.R:62]: `Styler` changed the syntax. The arguments and the function are the same.

**Comment by gowerc** [comment]
@G-Kazantzidis ,  Whats the relation between this PR and #58  ?

**Comment by G-Kazantzidis** [comment]
> @G-Kazantzidis , Whats the relation between this PR and #58 ?

This PR aims modify the treatment of the prior section. We want to take as an input a prior list such as
`priors <- list( "prior1" = "normal(0,1);", "prior2" = "normal(0,1);")` and produce a stan code that looks like 
`prior1 ~ normal(0,1); \n prior2 ~ normal(0,1); \n`. Thus, mainly adds "~" between the name and the definition. 
#58 aims to add an additional slot, "model",  in the StanModule object. Indeed the two PRs are connected.

**Comment by gowerc** [comment]
I guess I was wondering if they will conflict with each other or they are dependent on each other. As in was #58  branched off from this branch ?

**Comment by gowerc** [comment]
@G-Kazantzidis , Additionally would you mind resolving the merge conflicts with main please

**Pull request closed without merging**